x-pack/plugins/upgrade_assistant/public/components/tabs/checkup/deprecations/reindex/flyout/checklist_step.tsx

```diff
@@ -53,40 +80,64 @@ export const ChecklistFlyoutStep: React.StatelessComponent<{
-          title="Index is unable to ingest, update, or delete documents while reindexing"
+          title={
+            <FormattedMessage
+              id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.readonlyCallout.calloutTitle"
+              defaultMessage="Index is unable to ingest, update, or delete documents while reindexing"
+            />
+          }
           color="warning"
           iconType="alert"
         >
           <p>
-            If you can’t stop document updates or need to reindex into a new cluster, consider using
-            a different upgrade strategy.
+            <FormattedMessage
+              id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.readonlyCallout.cantStopDetail"
+              defaultMessage="If you can’t stop document updates or need to reindex into a new cluster,
+                consider using a different upgrade strategy."
+            />
           </p>
           <p>
-            Reindexing will continue in the background, but if Kibana shuts down or restarts you
-            will need to return to this page to resume reindexing.
+            <FormattedMessage
+              id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.readonlyCallout.backgroundResumeDetail"
+              defaultMessage="Reindexing will continue in the background, but if Kibana shuts down or restarts you will
+                need to return to this page to resume reindexing."
+            />
           </p>
         </EuiCallOut>
         {!hasRequiredPrivileges && (
           <Fragment>
             <EuiSpacer />
             <EuiCallOut
-              title="You do not have sufficient privileges to reindex this index"
+              title={
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.insufficientPrivilegeWarningCallout.title"
+                  defaultMessage="You do not have sufficient privileges to reindex this index"
+                />
+              }
               color="danger"
               iconType="alert"
             />
           </Fragment>
         )}
         <EuiSpacer />
         <EuiTitle size="xs">
-          <h3>Reindexing process</h3>
+          <h3>
+            <FormattedMessage
+              id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.reindexingChecklistTitle"
+              defaultMessage="Reindexing process"
+            />
+          </h3>
         </EuiTitle>
         <ReindexProgress reindexState={reindexState} cancelReindex={cancelReindex} />
       </EuiFlyoutBody>
       <EuiFlyoutFooter>
         <EuiFlexGroup justifyContent="spaceBetween">
           <EuiFlexItem grow={false}>
             <EuiButtonEmpty iconType="cross" onClick={closeFlyout} flush="left">
-              Close
+              <FormattedMessage
+                id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.closeButtonLabel"
```

Comment:
Curious, do we, or should we use a general key for common messages?

Reply:
I was advised by @Bamieh to not reuse ids for now. The id is supposed to closely reflect the UI hierarchy to assist translators in finding testing the usage of the label.

Reply:
👍 works for me.